enterprise/cmd/repo-updater/main.go

```diff
@@ -43,36 +40,18 @@ func enterpriseInit(
 	server *repoupdater.Server,
 ) (debugDumpers []debugserver.Dumper) {
 	ctx := context.Background()
-	campaignsStore := campaigns.NewStore(db)
-
-	syncRegistry := campaigns.NewSyncRegistry(ctx, campaignsStore, repoStore, cf)
-	if server != nil {
-		server.ChangesetSyncRegistry = syncRegistry
-	}
-
 	clock := func() time.Time {
 		return time.Now().UTC().Truncate(time.Microsecond)
 	}
 
-	sourcer := repos.NewSourcer(cf)
-	go campaigns.RunWorkers(ctx, campaignsStore, gitserver.DefaultClient, sourcer)
+	campaignsStore := campaigns.NewStoreWithClock(db, clock)
 
-	// Set up expired spec deletion
-	go func() {
-		for {
-			// We first need to delete expired ChangesetSpecs...
-			if err := campaignsStore.DeleteExpiredChangesetSpecs(ctx); err != nil {
-				log15.Error("DeleteExpiredChangesetSpecs", "error", err)
-			}
-			// ... and then the CampaignSpecs, due to the campaign_spec_id
-			// foreign key on changeset_specs.
-			if err := campaignsStore.DeleteExpiredCampaignSpecs(ctx); err != nil {
-				log15.Error("DeleteExpiredCampaignSpecs", "error", err)
-			}
+	syncRegistry := campaigns.NewSyncRegistry(ctx, campaignsStore, repoStore, cf)
+	if server != nil {
+		server.ChangesetSyncRegistry = syncRegistry
+	}
 
-			time.Sleep(2 * time.Minute)
-		}
-	}()
+	campaigns.SetupCampaignsBackgroundJobs(ctx, db, campaignsStore, repoStore, cf)
```

Comment:
In Go it's practice to consider the package name part of the name (e.g. `strings.Join`). I think this might be better:

```suggestion
	campaigns.StartBackgroundJobs(ctx, db, campaignsStore, repoStore, cf)
```

Reply:
To be more consistent with the goroutines, you could also just call it `MonitorBackgroundRoutines`.